src/core/mainloop/mainloop.c

```diff
@@ -2778,6 +2771,13 @@ do_main_loop(void)
    */
   initialize_periodic_events();
```

Comment:
So this is the only thing I'm confused and I would like to have documented why or resolved if we can.

These calls seems to belong to the subsystem initialization function where `initialize_periodic_events()` is actually invoked. If we can't use the initialize subsystem function for this, it kind of looses its purpose a bit to have two callsite that initialize. Seems very error prone to me over time :S.

Reply:
I've removed these calls, and nothing broke. :)